Check for `ipykernel` in New Project Wizard

sharon-wang opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://connect.posit.it/positron-wiki/pull-requests.html
* Ensure that the code is up-to-date with the `main` branch.
-->

### Intent

- Addresses https://github.com/posit-dev/positron/issues/2856

<img width="500" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/9c2dcd4d-3851-403a-90f6-65f21129b379">

https://github.com/posit-dev/positron/assets/25834218/7a07f721-7d1c-4487-851d-2ae489b81791

### Approach

- Add new command `python.isIpykernelInstalled` to the python extension
- Use the new command to check if the selected python interpreter has ipykernel installed
- Always show a note that `ipykernel` will be installed when a new environment is being created
- If an existing interpreter is being used, show a note that `ipykernel` will be installed if the selected interpreter doesn't have ipykernel is installed
- Fix an issue with the disabled `Next` button not maintaining disabled styling

### QA Notes

Does not include installing `ipykernel` during project creation. This will be added in a future PR.

sharon-wang committed:
add `ipykernelInstalled` to `PythonRuntimeExtraData`

sharon-wang committed:
change substep description to owner draw element or string

also use the ipykernel check in the python environment substep description/feedback

sharon-wang committed:
update project wizard height and substep styles

sharon-wang committed:
Revert "add `ipykernelInstalled` to `PythonRuntimeExtraData`"

This reverts commit 5774676808506291ef22a478669d9c03f93cdd29.

sharon-wang committed:
register command `python.isIpykernelInstalled` during `activatePositron`

This command checks if the interpreter corresponding to the provided Python path has a compatible ipykernel installed.

sharon-wang committed:
use `python.isIpykernelInstalled` in project wizard for ipykernel check

sharon-wang committed:
fix some comments and formatting

sharon-wang committed:
refactor wizard substep description and feedback

Instead of a function that returns either an element or string, pass the element or string directly for the PositronWizardSubStepProps description and feedback properties.

sharon-wang committed:
ensure next button remains disabled while hovering

sharon-wang committed:
move variable declaration to tighter scope

sharon-wang committed:
run prettier on `extensions/positron-python/src/client/positron/extension.ts`

seeM reviewed (commented):
Python extension changes LGTM!

sharon-wang committed:
check for runtimeStartupComplete before retrieving entries

sharon-wang committed:
use non-aliased `pythonPath` for `ipykernel` check

Use the pythonPath from the extraRuntimeData if it exists, otherwise use the runtimePath.
In particular, the pythonPath is needed to check if ipykernel is installed for Pyenv
environments (the runtimePath results in an error, although the pythonPath works fine,
seemingly because the runtimePath is aliased, i.e. starts with `~`). In many cases, the
pythonPath and runtimePath are the same. When they differ, it seems that the pythonPath
is the non-aliased path to the python interpreter.

sharon-wang committed:
clean up ipykernel check

sharon-wang committed:
check env setup type before checking for ipykernel

sharon-wang committed:
reword note on using pythonPath instead of runtimePath

timtmok reviewed (approved):
It worked for me! I did have to eventually restart VS Code before launching a dev Positron. It seemed to not picked up any new environments that I created with pyenv cli until I did that.
